Models/Pieces/Piece.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Chess_Backend.Models.Positions;
+
+namespace Chess_Backend.Models.Pieces
+{
+    public abstract class Piece
+    {
+
+        public abstract Color Color { get; }
+        public abstract Tile TilePosition { get; set; }
+
+        protected Piece()
```

Comment:
The constructor should be public and take the Color and Tile as arguments(set the class params once and never again)